### ISSUE TITLE
Bug in survey application while performing Ignore+Exit Changes in a Form

### ISSUE THREAD

**Issue by swaingotnochill** [opendatakit/tool-suite-X#230]
In survey application, when you create a instance in a form and try to exit it with the option "Ignore + Exit", then just a white popup window appears on top section and nothing happens. Ideally, it should just exit which doesn't happen and instead it gets stuck until you manually press the back button in device and then exit.

Steps to Reproduce:
1. Open a Form 
2. Create an instance.
3. Click on top left showing form name for menu to appear.
4. Click on Ignore+Exit option.

Below is scren recording of  reproduced error:
https://user-images.githubusercontent.com/54025913/113267187-ad789080-92f3-11eb-9fff-510b0edff49f.mp4

Survey v2.1.7

**Comment by linl33** [comment]
@swaingotnochill Did you see anything else wrong/weird in other parts of the survey? I'm unable to reproduce this issue.

**Comment by swaingotnochill** [comment]
@linl33 No, just the ignore changes+ exit does not work. I tested with different forms and "uninstall+reinstall" application, but it still remained the same.

**Comment by wbrunette** [comment]
@swaingotnochill please provide the Android version you are using. Also make sure your WebView has been updated in Google Play.

**Comment by emeka-okechukwu** [comment]
Hi @wbrunette, I also get the issue. I'm using the latest WebView version and my Android version is 10.0.

![IMAGE 2021-04-02 22:56:41](https://user-images.githubusercontent.com/81817744/113460039-e0f80f80-9406-11eb-816a-5d43d06090e6.jpg)

**Comment by linl33** [comment]
I'm not sure why this is happening. It works fine for me on the Android 10 emulator. @emeka-okechukwu-dev Does this happen on every form for you?

@wbrunette Can you try to reproduce this as well?

**Comment by emeka-okechukwu** [comment]
@linl33 yep happens on every form. I'm using Huawei as well.

**Comment by wbrunette** [comment]
@linl33 I tried to recreate on the emulator for Android 10 and failed to recreate it.. @swaingotnochill @emeka-okechukwu-dev can you try the Android emulator and see if you get the same result?

**Comment by wbrunette** [comment]
I also attempted to recreate the error on moto g7 play running Android 10 and failed. Though I was using the development branch.

**Comment by swaingotnochill** [comment]
@wbrunette the survey version is  2.1.7.I forgot to mention it.

**Comment by emeka-okechukwu** [comment]
Hi @wbrunette, I'll give it a try on my Android emulator and let you know.

Downloaded the app and currently on "Configuring ODK-X Survey".

**Comment by emeka-okechukwu** [comment]
Hi @wbrunette, got the same result on my emulator. Running version 2.1.7 rev 240

![Screenshot_1617870960](https://user-images.githubusercontent.com/81817744/113996010-49e7e900-9846-11eb-83cb-8c45c0745244.png)

**Comment by wbrunette** [comment]
Which specific emulator? Are you using the google play version?

**Comment by emeka-okechukwu** [comment]
@wbrunette I'm using the Android 10 emulator on Android Studio. Not the Google Play version but downloaded it from GitHub releases (https://github.com/odk-x/survey/releases).

**Comment by wbrunette** [comment]
@emeka-okechukwu-dev thanks for responding, the Android 10 emulator on Android Studio has 7 different emulators for Android 10. I am trying to figure out which one(s) you have installed and which one you used for the test so I can try and match it exactly. 
![Android10Emulator](https://user-images.githubusercontent.com/11892715/114227694-29915a80-992a-11eb-8c98-8d6e8ba7a5ca.PNG)

**Comment by emeka-okechukwu** [comment]
@wbrunette sorry for the late response.

<img width="373" alt="Screenshot 2021-04-12 at 20 18 49" src="https://user-images.githubusercontent.com/81817744/114456376-4d73cb00-9bcc-11eb-8096-8003dc503e04.png">


And when I click "view details" about my emulator

<img width="354" alt="Screenshot 2021-04-12 at 20 19 55" src="https://user-images.githubusercontent.com/81817744/114456492-772cf200-9bcc-11eb-87b1-09c9ee4cc015.png">

**Comment by emeka-okechukwu** [comment]
Hope these help?

**Comment by kenejane** [comment]
I would like to work on this issue

**Comment by wbrunette** [comment]
@kenejane I have assigned the issue to you

**Comment by kenejane** [comment]
Alright @wbrunette . I am looking into it already

**Comment by kenejane** [comment]
Hello @wbrunette , I downloaded the ODK-X survey version 2.1.8 and the pre-Release version 2.1.9 APK from the releases GitHub repo and I didn't encounter the issue described above. I ran the apps on my phone which operates on android version 9. Here is a link of how the app behaved when I replicated the issue. Do let me know if I should also use an emulator to replicate.

https://user-images.githubusercontent.com/28413136/139250775-9741ea69-f4f4-46d9-898b-924c63e10a6d.mp4

**Comment by wbrunette** [comment]
@kenejane Yes please try to use the exact emulator specified as the problem version by  emeka-okechukwu-dev

**Comment by kenejane** [comment]
Ok @wbrunette, let me work on it again

**Comment by kenejane** [comment]
Hello @wbrunette , The emulator on my android studio is not displaying when it's launched and it is because I don't have enough space on my computer for it to run seamlessly. However, I installed the ODK-X survey preview version 2.1.9 on my phone that runs on android 9 and I was able to reproduce the error encountered by @swaingotnochill . I didn't create a new issue because this was already open. Here is a demo of the same error reproduced

https://user-images.githubusercontent.com/28413136/139836322-0415c8d9-76b0-47fb-a883-73011d791d1e.mp4

**Comment by wbrunette** [comment]
@kenejane  thanks for confirming the problem was fixed. Please record this as a contribution for Outreachy
